include/ipr/impl

```diff
@@ -370,8 +370,8 @@ namespace ipr {
       struct Binding_capture_specification : Capture_specification<ipr::Capture_specification::Binding> {
          Binding_capture_specification(const ipr::Identifier& n, const ipr::Expr& x, Capture_mode m) : id{n}, init{x}, md{m} { }
          Capture_mode mode() const final { return md; }
-         const ipr::Identifier& name() const { return id; }
-         const ipr::Expr& initializer() const { return init; }
+         const ipr::Identifier& name() const final { return id; }
```

Comment:
Just a personal preference, but I prefer final on structs/classes as opposed to on methods.

Reply:
> Just a personal preference, but I prefer final on structs/classes as opposed to on methods.

I actually experimented with both (you can see some remnants in the existing codebase), and I've come down on systematically using `final` on the member functions: `final` on individual member function actually checks that such a declaration is overriding something from a base class.  `final` on a class does not achieve that.  Furthermore, `final` on individual member function is more flexible in terms of allowing further derivation (if needed) even if no overriding is intended.